xbmc/music/MusicUtils.cpp

```diff
@@ -444,6 +444,55 @@ class CAsyncGetItemsForPlaylist : public IRunnable
   CMusicDatabase m_musicDatabase;
 };
 
+SortDescription GetSortDescription(const CGUIViewState& state, const CFileItemList& items)
+{
+  SortDescription sortDescTrackNumber;
+
+  const auto& sortDescriptions = state.GetSortDescriptions();
+  for (const auto& sortDescription : sortDescriptions)
+  {
+    if (sortDescription.sortBy == SortByTrackNumber)
+    {
+      // check whether at least one item has actually a track number set
+      for (const auto& item : items)
+      {
+        if (item->HasMusicInfoTag() && item->GetMusicInfoTag()->GetTrackNumber() > 0)
+        {
+          // First choice for folders containing a single album
+          sortDescTrackNumber = sortDescription;
+          sortDescTrackNumber.sortOrder = SortOrderAscending;
+          break; // leave items loop. we can still find ByArtistThenYear. so, no return here.
+        }
+      }
+    }
+    else if (sortDescription.sortBy == SortByArtistThenYear)
+    {
+      // check whether songs from at least two different albums are in the list
+      int lastAlbumId = -1;
+      for (const auto& item : items)
+      {
+        if (item->HasMusicInfoTag())
+        {
+          const auto tag = item->GetMusicInfoTag();
+          if (lastAlbumId != -1 && tag->GetAlbumId() != lastAlbumId)
+          {
+            // First choice for folders containing multiple albums
+            SortDescription desc = sortDescription;
```

Comment:
Wouldn't it make more sense to make `sortDescTrackNumber` a bit more generic and just reuse it here as well just like you did above?
Or to simply remove the const modifier in the iterator and set the properties on the sortDescription itself?

Reply:
> Or to simply remove the const modifier in the iterator and set the properties on the sortDescription itself?

 Yes, will change that.

Reply:
done